jobsdb/unionQuery_test.go

```diff
@@ -111,3 +114,183 @@ func TestMultiTenantHandleT_GetAllJobs(t *testing.T) {
 	require.NoError(t, err, "Error getting All jobs")
 	require.Equal(t, 3, len(jobs.Jobs))
 }
+
+/*
+rt
+
+BenchmarkUnionQuery/UnionQuery-1000-8         	                    	                        1358249041 ns/op
+BenchmarkUnionQuery/Concurrent_Union_Query__-1000_-16-8         	                1666216834 ns/op
+BenchmarkUnionQuery/Concurrent_Union_Query__-1000_-32-8         	                3847614042 ns/op
+BenchmarkUnionQuery/ParallelQuery-1000-8                        	                                 1538826083 ns/op
+
+BenchmarkUnionQuery/UnionQuery-3000-8                           	                                 4177400417 ns/op
+BenchmarkUnionQuery/Concurrent_Union_Query__-3000_-16-8         	                 5335373167 ns/op
+BenchmarkUnionQuery/Concurrent_Union_Query__-3000_-32-8         	                 11321719167 ns/op
+BenchmarkUnionQuery/ParallelQuery-3000-8                        	                                 3538207958 ns/op
+
+BenchmarkUnionQuery/UnionQuery-5000-8                           	                                6365580875 ns/op
+BenchmarkUnionQuery/Concurrent_Union_Query__-5000_-16-8         	                8497499458 ns/op
+BenchmarkUnionQuery/Concurrent_Union_Query__-5000_-32-8         	                17646324750 ns/op
+BenchmarkUnionQuery/ParallelQuery-5000-8                        	                                 5540005584 ns/op
+
+BenchmarkUnionQuery/UnionQuery-10000-8                          	                                12698598542 ns/op
+BenchmarkUnionQuery/Concurrent_Union_Query__-10000_-16-8        	                19103076166 ns/op
+BenchmarkUnionQuery/Concurrent_Union_Query__-10000_-32-8        	                38923613417 ns/op
+BenchmarkUnionQuery/ParallelQuery-10000-8                       	                                 11816298834 ns/op
+
+BenchmarkUnionQuery/UnionQuery-100000-8                         	                                 143067570417 ns/op
+BenchmarkUnionQuery/Concurrent_Union_Query__-100000_-16-8       	                 200310106917 ns/op
+BenchmarkUnionQuery/Concurrent_Union_Query__-100000_-32-8       	                 409845463834 ns/op
+BenchmarkUnionQuery/ParallelQuery-100000-8                      	                                  118793139708 ns/op
+*/
+
+// BenchmarkUnionQuery is a benchmark for UnionQuery with 20 different workspaces
+// and jobSize is the total number of jobs split across the workspaces
+func BenchmarkUnionQuery(b *testing.B) {
```

Comment:
Can we add a benchmark where we partitition the table by workspaceID? I think it might give us some benefits in scenarios with many workspaces and parallel queries. We a configurable number of workspaces as per [this comment](https://github.com/rudderlabs/rudder-server/pull/2725/files#r1036208241) we should get an idea.

Reply:
I was thinking that we could try both `LIST` and `HASH` partitioning in the benchmark. With the `LIST` one it should fairly simple to setup a partition per workspace.